src/main/java/kitchenpos/table/application/OrderTableUngroupEventHandler.java

```diff
@@ -0,0 +1,47 @@
+package kitchenpos.table.application;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import kitchenpos.order.domain.OrderRepository;
+import kitchenpos.order.domain.OrderStatus;
+import kitchenpos.order.event.TableUngroupEvent;
+import kitchenpos.table.domain.OrderTable;
+import kitchenpos.table.domain.OrderTableRepository;
+import org.springframework.context.event.EventListener;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+@Component
+public class OrderTableUngroupEventHandler {
+
+    private final OrderRepository orderRepository;
+    private final OrderTableRepository orderTableRepository;
+
+    public OrderTableUngroupEventHandler(OrderRepository orderRepository, OrderTableRepository orderTableRepository) {
+        this.orderRepository = orderRepository;
+        this.orderTableRepository = orderTableRepository;
+    }
+
+    @Transactional
+    @EventListener
+    public void handle(TableUngroupEvent event) {
+        List<OrderTable> findOrderTables = orderTableRepository.findAllByTableGroupId(event.getTableGroupId());
+        List<Long> orderTableIds = getOrderTableIds(findOrderTables);
+        validateOrderStatus(orderTableIds);
+        findOrderTables.stream().forEach(OrderTable::detachFromTableGroup);
+    }
+
+    private void validateOrderStatus(List<Long> orderTableIds) {
+        if (orderRepository.existsByOrderTableIdInAndOrderStatusIn(orderTableIds,
+                Arrays.asList(OrderStatus.COOKING, OrderStatus.MEAL))){
```

Comment:
```suggestion
                Arrays.asList(OrderStatus.COOKING, OrderStatus.MEAL))) {
```

컨벤션을 위반하셨습니다. IntelliJ Auto Formatting 기능을 사용하는 습관을 들여보아요.

The auto-formatting shortcut in IntelliJ

- For windows Ctrl + Alt + L.
- For ubuntu Ctrl + Alt + windows + L.
- For Mac Command + Option + L.

Reply:
아이고 ㅠㅠ 좀더 습관적으로 코드 포매팅을 하겠습니다

Reply:
아이고 ㅠㅠ 아까 다시 리뷰 요청 할때 고친줄 알았는데 아니었네요 다시 포매팅 하여 수정 하였습니다!